### PR TITLE
[jit] Fill in more MONO_PATCH_INFO_NONE support.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -4002,16 +4002,17 @@ decode_patch (MonoAotModule *aot_module, MonoMemPool *mp, MonoJumpInfo *ji, guin
...
 	case MONO_PATCH_INFO_AOT_JIT_INFO:
 		ji->data.index = decode_value (p, &p);
 		break;
+	case MONO_PATCH_INFO_GC_SAFE_POINT_FLAG:
+	case MONO_PATCH_INFO_NONE:
+		break;
 	default:
-		g_warning ("unhandled type %d", ji->type);
-		g_assert_not_reached ();
+		g_error ("unhandled type %d", ji->type);
```
Comment (vargaz) [inline mono/mini/aot-runtime.c]: NONE should not be happen here.

Comment (jaykrell) [reply mono/mini/aot-runtime.c]: Oh, aot-compiler.c skips it? I'll check, and ok either way.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/mini-s390x.c [hunk 2]
```diff
@@ -5381,6 +5381,8 @@ mono_arch_patch_code (MonoCompile *cfg, MonoMethod *method, MonoDomain *domain,
 				target = S390_RELATIVE(target, ip);
 				ip += 2;
 				s390_patch_rel (ip, (guint64) target);
+			case MONO_PATCH_INFO_METHOD_NONE:
```
Comment (lambdageek) [inline mono/mini/mini-s390x.c]:
```suggestion
			case MONO_PATCH_INFO_NONE:
```

**Comment by lambdageek** [comment]
@monojenkins build

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**